### ISSUE TITLE
Can we upload data directky from browser to juicefs with minio at backend? (Also download data from juicefs to browser )

### ISSUE THREAD

**Issue by metawrufesh** [juicedata/juicefs#3746]
**What would you like to be added**:
I would like to upload and download file directly from browser to juicefs with minio at backend.
I see minio has the following feature:
https://min.io/docs/minio/linux/developers/python/API.html#presigned_put_object
https://min.io/docs/minio/linux/developers/python/API.html#presigned_get_object

May be we cannot use s3 gateway because it that way it may not be direct upload.

**Why is this needed**:
I have a use case of uploading large data from a web which can be further used for both download and as input for some model.

Please kindly put some suggestions.

**Comment by solracsf** [comment]
Using WebDAV maybe?
https://juicefs.com/docs/community/deployment/webdav/

**Comment by metawrufesh** [comment]
@solracsf Thank you so much for your suggestion. 

Looks like webDAV also acts like gateway, so at the end the file will first reach to webdav and then to s3 storage. 

I was wondering  may be if  we can have a juicefs interface on browser that can directly write to s3  storage in juicefs format (in slices and chunks).

**The reason I want to do this are:** 
The upload has to be made from browser as the user are not familiar with **CUI interface to use terminal s3 client** _( client using juicefs s3 gateway )._ 

One way is to upload a file to my server and then put it to s3 storage via **juicefs s3 gateway**. But I think that would be quite slow with large files.

**Comment by SandyXSD** [comment]
The juicefs S3 gateway should be simple to use as you only need to drag a file to the browser to upload it.

Another alternative is to mount juicefs on your server, and use the mount point as data directory for your service. Then any file uploaded will be written to juicefs directly.

**Comment by metawrufesh** [comment]
@SandyXSD Thank you for your reply.

> Not sure if you were asking for a browser extension (not supported)?

Yes I was wondering about the similar browser client that can directly write to s3 storage without without making a hop to gateway server. I think authentication to s3 storage and dbms system could be an issue if we want to implement this. 

> The juicefs S3 gateway should be simple to use as you only need to drag a file to the browser to upload it.

Not sure how this works. Do we also get a web interface with s3 gateway?

**Comment by SandyXSD** [comment]
Yes, see: https://juicefs.com/docs/community/s3_gateway/#quickstart
